classes/03class/exercises/c03-aws01/alanlima/README.md

```diff
@@ -0,0 +1,210 @@
+# C03-aws01
+
+## Attachments
+
+- [User Data Script](user-data.txt)
+
+## Command Execution Output
+
+- Commands used to create your autoscaling group:
+
+    ```bash
+    # Create a new Security group to allow connection to port 80
+    $ aws ec2 create-security-group \
+        --group-name c03-aws01-sg \
+        --description "Allow connection to port 80" \
+        --vpc-id vpc-08fb60055e82efe81 \
+        --tag-specifications "ResourceType=security-group,Tags=[{Key=Name,Value=c03-aws01-sg},{Key=DevopsClass,Value=c03}]"
+
+    GroupId: sg-07739f756a518e82e
+    Tags:
+    - Key: Name
+    Value: c03-aws01-sg
+    - Key: DevopsClass
+    Value: c03
+
+    # Update the security group to allow ingress from anywhere to port 80
+    $ aws ec2 authorize-security-group-ingress \
+        --group-id sg-07739f756a518e82e \
+        --cidr 0.0.0.0/0 \
+        --port 80 \
+        --protocol tcp
+
+    # Create the Launch Configuration resource with the template details to be applied for each instance when created
+    $ aws autoscaling create-launch-configuration \
+        --launch-configuration-name devops-init-instance-data \
+        --user-data file://user-data.txt \
+        --image-id ami-088ff0e3bde7b3fdf \
+        --instance-type t2.micro \
+        --security-groups sg-07739f756a518e82e
+        
+    # Describe the launch configuration recent created
+    $ aws autoscaling describe-launch-configurations \
+        --launch-configuration-names devops-init-instance-data
+    ```
+
+    ```yaml
+    LaunchConfigurations:
+    - BlockDeviceMappings: []
+    ClassicLinkVPCSecurityGroups: []
+    CreatedTime: '2020-07-08T22:04:09.980000+00:00'
+    EbsOptimized: false
+    ImageId: ami-088ff0e3bde7b3fdf
+    InstanceMonitoring:
+        Enabled: true
+    InstanceType: t2.micro
+    KernelId: ''
+    KeyName: ''
+    LaunchConfigurationARN: arn:aws:autoscaling:ap-southeast-2:097922957316:launchConfiguration:57f72e70-835b-40d7-9217-53f150c72e91:launchConfigurationName/devops-init-instance-data
+    LaunchConfigurationName: devops-init-instance-data
+    RamdiskId: ''
+    SecurityGroups:
+    - sg-07739f756a518e82e
+    UserData: IyEgL2Jpbi9iYXNoCnl1bSB1cGRhdGUgLXkKeXVtIGluc3RhbGwgLXkgaHR0cGQKY3VybCAxNjkuMjU0LjE2OS4yNTQvbGF0ZXN0L21ldGEtZGF0YS9ob3N0bmFtZSA+IGluZGV4Lmh0bWwKbXYgaW5kZXguaHRtbCAvdmFyL3d3dy9odG1sLwpzeXN0ZW1jdGwgc3RhcnQgaHR0cGQ=
+    ```
+
+    ```bash
+    # Now we can create the auto scaling group using the launch configuration created
+    $ aws autoscaling create-auto-scaling-group \
+        --auto-scaling-group-name asg-devops \
+        --min-size 1 --max-size 2 --desired-capacity 2 \
+        --launch-configuration-name devops-init-instance-data \
+        --vpc-zone-identifier "subnet-068860ac143dd2beb" \
```

Comment:
Please use at least 2 subnets (AZs), as per `Subnet: any two public subnet` in order for your solution to be high available

Reply:
I missed that 😅
I update the commands to include that. I also checked that the ASG placed the instances correctly in two different zone. 🎉